src/Randousel/randousel.svelte

```diff
@@ -0,0 +1,99 @@
+<script>
+  import { Card, CardSubtitle, CardTitle, CardText, CardActions, Button, MaterialApp } from 'svelte-materialify';
+  import { format } from 'd3';
+
+  export let location;
+  export let wtfData;
+  let count = 0;
+  let data = []; // Initialize data
+  let selectedIndex = 0; // Randomly pick an array element
+  let selectedWTF;
+
+  const loading = Promise.all([location, wtfData])
```

Comment:
Other than the `await` mechanism being used in the HTML to show the "now loading" text, I'm not sure much of this needs to be in the component itself because it's not influenced by anything in here. Could move the loading stuff out to `graphic.js` and pass in the ready to go `data` and `location` instead.

Then the "now loading" disclaimer could move to the HTML itself — once Svelte takes over the container it's gonna clear it out and you'll essentially have the same effect (with a simpler component)!

Reply:
So there's no penalty for having markup inside the component before it mounts???